projects/packages/search/src/dashboard/components/pages/sections/plan-usage-section.jsx

```diff
@@ -1,26 +1,52 @@
 import { ContextualUpgradeTrigger, ThemeProvider } from '@automattic/jetpack-components';
 import { createInterpolateElement } from '@wordpress/element';
-import { __, sprintf } from '@wordpress/i18n';
+import { __ } from '@wordpress/i18n';
 import React from 'react';
 import DonutMeterContainer from '../../donut-meter-container';
+import PlanSummary from './plan-summary';
 
 // import './plan-usage-section.scss';
 
-// TODO: Replace local PlanSummary component with new component when ready.
-const PlanUsageSection = props => {
-	if ( ! props.isVisible ) {
+const usageInfoFromAPIData = apiData => {
+	// Transform the data as necessary.
+	// Are there better defaults for the Max values?
+	// Should we recored, log, or otherwise surface potential errors here?
+	return {
+		recordCount: apiData?.currentUsage?.num_records || 0,
+		recordMax: apiData?.currentPlan?.record_limit || 0,
+		requestCount: apiData?.latestMonthRequests?.num_requests || 0,
+		requestMax: apiData?.currentPlan.monthly_search_request_limit || 0,
+	};
+};
+
+const upgradeTypeFromAPIData = apiData => {
+	// Determine if upgrade message is needed.
+	if ( ! apiData.currentUsage.must_upgrade ) {
 		return null;
 	}
+	// Determine appropriate upgrade message.
+	let mustUpgradeReason = '';
+	if ( apiData.currentUsage.upgrade_reason.requests ) {
+		mustUpgradeReason = 'requests';
+	}
+	if ( apiData.currentUsage.upgrade_reason.records ) {
+		mustUpgradeReason = mustUpgradeReason === 'requests' ? 'both' : 'records';
+	}
+	return mustUpgradeReason;
+};
+
+const PlanUsageSection = props => {
 	// TODO: Add logic for plan limits.
-	const upgradeMessage = null;
+	const upgradeType = upgradeTypeFromAPIData( props.planInfo );
+	const usageInfo = usageInfoFromAPIData( props.planInfo );
```

Comment:
It might be more appropriate to perform API data transformation in our data layer (i.e., [our Redux state](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/search/src/dashboard/store)), enabling unit testing of such transformations.

Reply:
I'll look into this!